block/manager.go

```diff
@@ -249,6 +243,56 @@ func (m *Manager) AggregationLoop(ctx context.Context, lazy bool) {
 	}
 }
 
+func (m *Manager) SetFraudProofService(fraudProofServ *fraudserv.ProofService) {
+	m.executor.SetFraudProofService(fraudProofServ)
+}
+
+func (m *Manager) ProcessFraudProof(fraudProofService *fraudserv.ProofService, ctx context.Context, cancel context.CancelFunc) {
```

Comment:
I see this code is being duplicated across light node and block manager. Can we refactor/consolidate this code to avoid that?

Reply:
i replicated some code because of the validation (`VerifyFraudProof`) calls and also the way full node reacts (gracefully halt by canceling the context) is different from light node (which panic). 

in future we may have very less code to process the fraud proof as most of the logic will be moved to Proof.Validate function.